bundles/org.openhab.binding.mielecloud/README.md

```diff
@@ -0,0 +1,648 @@
+| ----------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------ |
+| locale      | mandatory | The locale to use for full text channels of things from this account. Possible values are `en`, `de`, `da`, `es`, `fr`, `it`, `nl`, `nb`. Default is `en`. | Prior to the 7th January 2021 `da`, `es`, `fr`, `it`, `nl` and `nb` will default to English. |
+
+Example things-file:
+```
+Bridge mielecloud:account:home [ locale="en" ]
+```
+
+### Appliance Configuration
+
+The binding configuration UI will show a things-file template containing things for all supported appliances from the paired account.
+This can be used as a starting point for a custom things-file.
+All Miele cloud appliance things have no configuration parameters.
+
+Example things-file:
+```
+Bridge mielecloud:account:home [ locale="en" ] {
+    Thing coffee_system 000703261234 "Coffee machine CVA7440" [ ]
+    Thing hob 000160102345 "Cooktop KM7677" [ ]
+    Thing washing_machine 000148503456 "Washing Machine WWV980" [ ]
+    Thing oven 000137434567 "Oven" [ ]
+    Thing fridge 007109235678 "Refrigerator K34483" [ ]
+    Thing dishwasher 000159456789 "Dishwasher G5000" [ ]
+    Thing fridge_freezer 711397890 "Fridge-freezer" [ ]
+    Thing dryer 000091098901 "Dryer" [ ]
+}
+```
+
+## Channels
+
+The following table lists all available channels.
+See the following chapters for detailed information about which appliance supports which channels.
+Depending on the exact appliance configuration not all channels might be supported, e.g. a hob with four plates will only fill the channels for plates 1-4.
+
+| Channel Type ID               | Item Type            | Description                                                                                                                               | Read only |
+| ----------------------------- | -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------- | --------- |
+| remote_control_can_be_started | Switch | Indicates if this device can be started remotely. | Yes |
+| remote_control_can_be_stopped | Switch | Indicates if this device can be stopped remotely. | Yes |
+| remote_control_can_be_paused | Switch | Indicates if this device can be paused remotely. | Yes |
+| remote_control_can_be_switched_on | Switch | Indicates if the device can be switched on remotely. | Yes |
+| remote_control_can_be_switched_off | Switch | Indicates if the device can be switched off remotely. | Yes |
+| remote_control_can_set_program_active | Switch | Indicates if the active program of the device can be set remotely. | Yes |
+| spinning_speed | String | The spinning speed of the active program. | Yes |
+| spinning_speed_raw | Number | The raw spinning speed of the active program. | Yes |
```

Comment:
Did you take a look at Channel [state descriptions](https://www.openhab.org/docs/developer/bindings/thing-xml.html#state-description)? That might make the _raw Channels unnecessary.

Reply:
I assume you are referring to the `<options>` tag specifically, but using this is in our opinion not reasonably possible at the moment. In order to provide a full description of the possible options the list would have to be either static or accessible via the cloud. Unfortunately, both is not the case. I will use the active program / program ID as an example but the following outline applies to all channels that we defined raw channels for.

The list of possible options for the active program depends on the kind of device (oven, washing machine, ...) and on the specific model. Additionally, for some devices custom programs can be defined. Thus, we cannot statically list all possible values in the binding.

Also, there is no endpoint in the Miele 3rd Party API that supplies a list of possible values including translations. The only supplied information is the localized value of the currently selected / active program. We considered to dynamically alter the channel with these values once they are received but found it to be rather confusing than helpful from a user's perspective.

Have other bindings had similar problems? How did they solve them? Or is our solution fine given these circumstances? If you have a good advice we will happily apply it.

Reply:
If you don't have a mapping of the options, it's barely possible to use state descriptions. I don't have any better solution.